src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -429,7 +429,8 @@ private boolean shouldRotate(long now) {
 
   private void readOffset() throws ConnectException {
     try {
-      String path = FileUtils.topicDirectory(url, topicsDir, tp.topic());
+      String partitionedPath = partitioner.generatePartitionedPath(tp.topic(), "");
```

Comment:
This is better than current state, but technically incorrect as `generatedPartitionedPath` might require the partition to produce a working output.